pkg/storage/pebble_iterator.go

```diff
@@ -892,10 +892,11 @@ func (p *pebbleIterator) destroy() {
 		panic("iterator still in use")
 	}
 	if p.iter != nil {
-		err := p.iter.Close()
-		if err != nil {
-			panic(err)
-		}
+		_ = p.iter.Close()
+		// TODO(jackson): The error returned by iter.Close() may be an ephemeral
```

Comment:
This should call out that if `iter.Valid()` returns an error, then that same error will also be returned by `Close()`. In most cases, we do not care about that error during `Close`, since the caller may already have handled it appropriately (e.g. a context cancellation) but still wants to close the iterator via e.g. `defer`.